skyreader/result.py

```diff
@@ -21,12 +21,36 @@
 import numpy as np
 import pandas as pd
 from astropy.io import ascii
+from matplotlib import patheffects
 from matplotlib import pyplot as plt
-from matplotlib import text, patheffects
+from matplotlib import text
 
 from .event_metadata import EventMetadata
+from .plot.plotting_tools import (
+    DecFormatter,
+    RaFormatter,
+    format_fits_header,
+    hp_ticklabels,
+    plot_catalog,
+)
+
+###############################################################################
+# CONSTANTS
+
+
+# bookkeeping for comparing values
+DEFAULT_RTOL_PER_FIELD = {  # w/ rtol values
+    # any field not here is assumed to require '==' for comparison
+    "llh": 1e-4,
+    "E_in": 1e-2,
+    "E_tot": 1e-2,
+}
+CANNOT_BE_ZERO_FIELDS = [
+    # if field's val is 0, then all the pixel's numerical datapoints are "isclose"
+    "E_in",
+    "E_tot",
+]
```

Comment:
I believe the idea is that a pixel with a zero energy is an invalid _value_, but that alone should not be enough to make `is_close()` false. Now, if `do_disqualify_zero_energy_pixels` is True, then the entire _pixel_ is considered invalid and, again, it is not enough to make `is_close()` false.

That's what the code says at least. I'm not sure if we want to keep it (hence all the TODOs). 

If I remember correctly, `millipede_original` has a bug where low likelihoods produce a zero energy value.

Reply:
We're using `--disqualify-zero-energy-pixels` in the scanner's CI tests

Reply:
OK, I think I am just confused because:
1. the wording _all the pixel's numerical datapoints are "isclose"_ seems to imply that a zero value here would make automatically the pixel pass the "isclose" criterion, regardless of other fields such as the `llh` value;
2. the `CANNOT_BE_ZERO` naming seems to imply, on the contrary, a more restrictive criterion where the pixel is never "close".

I cannot suggest a better naming / comment without taking a deeper look at the code, though. I am fine leaving it as it is now for the time being.

Reply:
I agree the naming is bad -- I'll try some renaming...

Reply:
> OK, I think I am just confused because:
> 
> 1. the wording _all the pixel's numerical datapoints are "isclose"_ seems to imply that a zero value here would make automatically the pixel pass the "isclose" criterion, regardless of other fields such as the `llh` value;

When `do_disqualify_zero_energy_pixels` is True, that's how it behaves. I think we could remove `do_disqualify_zero_energy_pixels` and make one standard for how we treat this. Which would you prefer?

Reply:
> > OK, I think I am just confused because:
> > 
> > 1. the wording _all the pixel's numerical datapoints are "isclose"_ seems to imply that a zero value here would make automatically the pixel pass the "isclose" criterion, regardless of other fields such as the `llh` value;
> 
> When `do_disqualify_zero_energy_pixels` is True, that's how it behaves. I think we could remove `do_disqualify_zero_energy_pixels` and make one standard for how we treat this. Which would you prefer?

Intuitively, I would say that if one of those fields is zero then the pixel `isclose` should still require the other fields to be `isclose`. But I cannot remember if this was causing problems at the time.

Reply:
Good point, testing it here: https://github.com/icecube/skymap_scanner/pull/198

Reply:
see #10